test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,173 @@
  * @bug 8290313
  * @library /test/lib
  * @summary Produce warning when user specified java.io.tmpdir directory doesn't exist
+ * @run junit TempDirDoesNotExist
  */
 
-import jdk.test.lib.process.ProcessTools;
 import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
 
 import java.io.File;
+import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-public class TempDirDoesNotExist {
-    final static String ioWarningMsg = "WARNING: java.io.tmpdir directory does not exist";
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
 
-    public static void main(String... args) throws Exception {
+public class TempDirDoesNotExist {
+    final static String WARNING = "WARNING: java.io.tmpdir directory does not exist";
 
-        String userDir = System.getProperty("user.home");
-        String timeStamp = System.currentTimeMillis() + "";
-        String tempDir = Path.of(userDir,"non-existing-", timeStamp).toString();
+    private static final String USER_DIR = System.getProperty("user.home");
 
+    public static void main(String... args) throws IOException {
```

Comment:
Might be worth a comment that this is spawned to test combinations of parameters.

Reply:
Comment added in cab8fa50a848f3c796bc81adcfd214aef0d054d3.